src/nvim/drawline.c

```diff
@@ -1098,6 +1189,12 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
     extra_check = true;
   }
 
+  statuscol_T statuscol = { 0 };
+  if (*wp->w_p_stc != NUL && (wp->w_p_nu || wp->w_p_rnu)) {
```

Comment:
Since this option has evolved from being a 'numberformat', I think it no longer makes sense to only draw the statuscolumn if either 'number' or 'relativenumber' is set. Will remove.